.github/workflows/release-client.yml

```
@@ -0,0 +1,342 @@
+name: Release Client Build
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  workflow_dispatch:
+jobs:
+  checks-and-tests:
+    runs-on: [self-hosted, Linux, X64]
+    steps:
+      - name: Free disk space
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "/usr/local/share/boost"
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+          df -h
+
+      - name: Checkout the source code
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Install deps
+        run: sudo apt -y install protobuf-compiler
+
+      - name: Install & display rust toolchain
+        run: rustup show
+
+      - name: Check targets are installed correctly
+        run: rustup target list --installed
+
+      - name: Install cargo-nextest
+        run: curl -LsSf https://get.nexte.st/latest/linux | tar zxf - -C ${CARGO_HOME:-~/.cargo}/bin
+
+      - name: Check all features compilation
+        run: cargo check --features try-runtime,runtime-benchmarks --locked
+
+      - name: Run all tests
+        run: make test-all
+
+  native-linux:
+    needs: checks-and-tests
+    runs-on: [self-hosted, Linux, X64]
+    strategy:
+      matrix:
+        target:
+          - x86_64-unknown-linux-gnu
+          - aarch64-unknown-linux-gnu
+
+    steps:
+      - name: Checkout the source code
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Install deps
+        run: sudo apt -y install protobuf-compiler lld
+
+      - name: aarch64 setup
+        if: contains(matrix.target, 'aarch64')
+        shell: bash
+        run: |
+          sudo apt update
+          sudo apt install -y gcc-multilib g++-multilib
+          sudo apt install -y gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
+
+          mkdir -p .cargo
+          touch .cargo/config
+          printf '[target.aarch64-unknown-linux-gnu]\nlinker = "aarch64-linux-gnu-gcc"' >> .cargo/config
+
+      - name: x86_64 setup
+        if: contains(matrix.target, 'x86_64')
+        run: |
+          mkdir -p .cargo
+          touch .cargo/config
+          printf '[target.x86_64-unknown-linux-gnu]\nrustflags = ["-Clink-arg=-fuse-ld=lld"]' >> .cargo/config
+
+      - name: Install & display rust toolchain
+        run: rustup show
+
+      - name: Add aarch64 target
+        if: contains(matrix.target, 'aarch64')
+        run: rustup target add ${{ matrix.target }}
+
+      - name: Check targets are installed correctly
+        run: rustup target list --installed
+
+      - name: Build optimized binary
+        run: cargo build --profile production --target ${{ matrix.target }} --locked
```

```suggestion
        run: cargo build --profile production --target ${{ matrix.target }} --locked --bin astar-collator
```
to skip a few packages like integrations-tests from building